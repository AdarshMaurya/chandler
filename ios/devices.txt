Device ID	Device Name
d988d298307e29bd646dca1f58c14825cf5a82bd	Theo's Iphone SE
