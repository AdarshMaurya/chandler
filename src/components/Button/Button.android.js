import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {
  TouchableNativeFeedback,
  View,
  Text,
  ActivityIndicator,
} from 'react-native';
import {styles} from './Button.styles';

export class Button extends Component {
  static defaultProps = {
    disabled: false,
    loading: false,
  };

  static propTypes = {
    onPress: PropTypes.func.isRequired,
    title: PropTypes.string,
    children: PropTypes.string,
    disabled: PropTypes.bool,
    loading: PropTypes.bool,
    loadingTitle: PropTypes.string,
  };

  getTitle() {
    const {title, children, loadingTitle, loading} = this.props;
    if (loading && loadingTitle) {
      return loadingTitle.toUpperCase();
    }
    return (title || children || '').toUpperCase();
  }

  render() {
    const {disabled, loading} = this.props;

    return (
      <TouchableNativeFeedback
        onPress={this.props.onPress}
        disabled={disabled}
        background={TouchableNativeFeedback.SelectableBackground()}>
        <View style={[styles.button, disabled && styles.buttonDisabled]}>
          <Text style={styles.buttonText}>{this.getTitle()}</Text>
          {loading && (
            <ActivityIndicator
              color="#ffffff"
              style={styles.activityIndicator}
            />
          )}
        </View>
      </TouchableNativeFeedback>
    );
  }
}
