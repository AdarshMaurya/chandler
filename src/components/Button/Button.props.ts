export interface IButtonProps {
  onPress: (...args) => any;
  children?: string;
  loading?: boolean;
  title?: string;
  disabled?: boolean;
  loadingTitle?: string;
}
