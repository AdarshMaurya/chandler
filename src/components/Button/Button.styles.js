import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  button: {
    height: 45,
    borderRadius: 5,
    marginVertical: 10,
    backgroundColor: 'rgb(43, 187, 72)',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    flexDirection: 'row',
  },
  activityIndicator: {
    marginLeft: 10,
  },
  buttonDisabled: {
    backgroundColor: 'rgba(43, 187, 72, 0.4)',
  },
  buttonText: {
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
  },
});
