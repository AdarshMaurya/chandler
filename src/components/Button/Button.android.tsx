import React, {PureComponent} from 'react';
import {
  ActivityIndicator,
  Text,
  TouchableNativeFeedback,
  View,
} from 'react-native';

import {IButtonProps} from './Button.props';
import {styles} from './Button.styles';

export class Button extends PureComponent<IButtonProps> {
  public static defaultProps = {
    disabled: false,
    loading: false,
  };

  public getTitle() {
    const {title, children, loadingTitle, loading} = this.props;
    if (loading && loadingTitle) {
      return loadingTitle.toUpperCase();
    }
    return (title || children || '').toUpperCase();
  }

  public render() {
    const {disabled, loading} = this.props;

    return (
      <TouchableNativeFeedback
        onPress={this.props.onPress}
        disabled={disabled}
        background={TouchableNativeFeedback.SelectableBackground()}
      >
        <View style={[styles.button, disabled && styles.buttonDisabled]}>
          <Text style={styles.buttonText}>{this.getTitle()}</Text>
          {loading && (
            <ActivityIndicator
              color="#ffffff"
              style={styles.activityIndicator}
            />
          )}
        </View>
      </TouchableNativeFeedback>
    );
  }
}
