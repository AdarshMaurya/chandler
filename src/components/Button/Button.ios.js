import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text } from 'react-native';
import { styles } from './Button.styles';


export class Button extends Component {
  static propTypes = {
    onPress: PropTypes.func,
    text: PropTypes.string,
    children: PropTypes.string,
  }

  getText () {
    return this.props.text || this.props.children || '';
  }

  render () {
    return (
      <TouchableOpacity style={styles.button} onPress={this.props.onPress}>
        <Text style={styles.buttonText}>{this.getText()}</Text>
      </TouchableOpacity>
    );
  }
}
