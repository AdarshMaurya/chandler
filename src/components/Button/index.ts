import {Platform} from 'react-native';
import {Button as ButtonAndroid} from './Button.android';
import {Button as ButtonIOS} from './Button.ios';

export const Button = Platform.OS === 'ios' ? ButtonIOS : ButtonAndroid;
