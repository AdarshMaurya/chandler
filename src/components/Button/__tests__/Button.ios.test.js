import React from 'react';
import {shallow} from 'enzyme';
import {Button} from '../Button.ios';

describe('Components', () => {
  describe('Button', () => {
    const defaultOnPress = {
      onPress: () => {},
    };

    it('should renders correctly', () => {
      expect(
        shallow(<Button {...defaultOnPress}>Test</Button>),
      ).toMatchSnapshot();
      expect(
        shallow(<Button {...defaultOnPress} title="qdwdqw" />),
      ).toMatchSnapshot();
    });

    it('should contain the same title than passed into parameter', () => {
      const button1 = shallow(<Button {...defaultOnPress} title="Test" />);
      expect(button1.find('Text').children().text()).toEqual('Test'); // eslint-disable-line

      const button2 = shallow(<Button {...defaultOnPress} title="Test2" />);
      expect(button2.find('Text').children().text()).toEqual('Test2'); // eslint-disable-line
    });

    it('should call onPress when taping the button', () => {
      const onPress = jest.fn();
      const button = shallow(<Button onPress={onPress} />);
      button.find('TouchableOpacity').prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
