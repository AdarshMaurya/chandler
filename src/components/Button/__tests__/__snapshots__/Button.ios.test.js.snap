// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Button should renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Test
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Test
      </Text>,
      "onPress": undefined,
      "style": Object {
        "backgroundColor": "#2BBB48",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Test",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "Test",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Test
        </Text>,
        "onPress": undefined,
        "style": Object {
          "backgroundColor": "#2BBB48",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Test",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Test",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components Button should renders correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="qdwdqw"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        qdwdqw
      </Text>,
      "onPress": undefined,
      "style": Object {
        "backgroundColor": "#2BBB48",
        "borderRadius": 5,
        "height": 45,
        "justifyContent": "center",
        "marginVertical": 10,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "qdwdqw",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "qdwdqw",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          qdwdqw
        </Text>,
        "onPress": undefined,
        "style": Object {
          "backgroundColor": "#2BBB48",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 10,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "qdwdqw",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "qdwdqw",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
